### PR TITLE
FakeTensorMode and Prims.add/sub/mul/div support scalar only inputs

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87759

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

**Review by mruberry — APPROVED** [review]
prim and test updates look correct to me!

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -17392,14 +17392,14 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
...
-        supports_two_python_scalars=False,
+        supports_two_python_scalars=True,
         supports_one_python_scalar=True,
     ),
     ElementwiseBinaryPythonRefInfo(
         "_refs.true_divide",
         torch_opinfo_name="true_divide",
         # https://github.com/pytorch/pytorch/issues/76944
-        supports_two_python_scalars=False,
+        supports_two_python_scalars=True,
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:17402]: If these were not explicitly specified would the values be inherited?

File: torch/_subclasses/fake_tensor.py [hunk 2]
```diff
@@ -870,13 +877,25 @@ def gen_wrap_fn(self, func, args, kwargs):
...
                 if common_device is None:
-                    common_device = FakeTensor._find_common_device(func, args, kwargs)
-                return converter(self, e, device or common_device)
+                    (
+                        common_device,
+                        has_scalar_only_inputs,
+                    ) = FakeTensor._find_common_device(func, args, kwargs)
+
+                if has_scalar_only_inputs:
+                    # Under FakeTensorMode, op accepts scalar only inputs, such as aten.add/sub/mul/div,
```
Comment (eellison) [inline torch/_subclasses/fake_tensor.py:893]: It's a bit strange that previously with meta in the tls, the output would be a meta tensor, and now it outputs a normal cpu tensor.. but probably not worth diving deep into.

**Review by eellison — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (SherlockNoMad) [reply torch/testing/_internal/common_methods_invocations.py:17436]: default value for supports_two_python_scalars is False

**Pull request closed without merging**